search/index.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	api "github.com/tigrisdata/tigris-client-go/api/server/v1"
+
+	"github.com/tigrisdata/tigris-client-go/driver"
+	"github.com/tigrisdata/tigris-client-go/filter"
+	"github.com/tigrisdata/tigris-client-go/schema"
+)
+
+var ErrNotFound = fmt.Errorf("document not found")
```

Comment:
is it possible to return HTTP 404 with a message here?

Reply:
How it would be useful?
In Go user should check the error like this:
```
if errors.Is(err, tigris.ErrNotfound) {
...
}
```
It's abstract from protocol implementation.

Reply:
I noticed in gotrue this error was getting leaked as an internal server error because the user may not be aware of this error unless obviously, we have someplace to look for all these errors or helper methods to know what happened.

But this is not related to this diff, in general about error handling if we need to do anything.

Reply:
If it's leaked, it's a bug an should be fixed.
But the idea is to use driver.Error on the driver level and tigris.Error on the high level.
Here is usage examples:
Driver: https://github.com/tigrisdata/tigris-client-go/blob/main/driver/example_test.go#L63
Client: https://github.com/tigrisdata/tigris-client-go/blob/main/tigris/example_test.go#L133

Reply:
I've got what you are saying. Fixed.